frontend/javascripts/oxalis/view/right-border-tabs/starting_job_modals.js

```diff
@@ -0,0 +1,292 @@
+  return (
+    <StartingJobModal
+      dataset={dataset}
+      userBoundingBoxes={userBoundingBoxes}
+      handleClose={handleClose}
+      jobName="nuclei inferral"
+      jobApiCall={colorLayerName =>
+        startNucleiInferralJob(dataset.owningOrganization, dataset.name, colorLayerName)
+      }
+      description={
+        <>
+          <p>
+            Start a job that automatically detects nuclei for this dataset. This job creates a copy
+            of this dataset once it has finished. The new dataset will contain the detected nuclei
+            as a segmentation layer.
+          </p>
+          <p>
+            <b>
+              Note that this feature is still experimental. Nuclei detection currently works best
+              with EM data and a resolution of approximately 200{ThinSpace}nm per voxel. The
+              inferral process will automatically use the magnification that matches that resolution
+              best.
+            </b>
+          </p>
+        </>
+      }
+    />
+  );
+}
+
+function _NeuronInferralModal({ dataset, handleClose, userBoundingBoxes }: Props) {
+  return (
+    <StartingJobModal
+      dataset={dataset}
+      handleClose={handleClose}
+      userBoundingBoxes={userBoundingBoxes}
+      jobName="neuron inferral"
+      isBoundingBoxConfigurable
+      jobApiCall={async (colorLayerName, boundingBox) => {
+        if (!boundingBox) {
+          return Promise.resolve();
+        }
+        const bbox = computeArrayFromBoundingBox(boundingBox.boundingBox);
+        return startNeuronInferralJob(
+          dataset.owningOrganization,
+          dataset.name,
+          colorLayerName,
+          bbox,
+        );
+      }}
+      description={
+        <>
+          <p>
+            Start a job that automatically detects the neurons for this dataset. This job creates a
+            copy of this dataset once it has finished. The new dataset will contain the new
+            segmentation which segments the neurons of the dataset.
+          </p>
+          <p>
+            <b>
+              Note that this feature is still experimental and takes quite a long time. Thus we
+              suggest to use a small bounding box and not the full dataset extent. The neuron
+              detection currently works best with EM data. The best resolution for the process will
+              be chosen automatically.
+            </b>
+          </p>
+        </>
+      }
+    />
+  );
+}
+
+function mapStateToProps(state: OxalisState): StateProps {
+  const userBB = getUserBoundingBoxesFromState(state);
+  return {
+    dataset: state.dataset,
+    userBoundingBoxes: userBB,
+  };
+}
+
+export const NeuronInferralModal = connect<Props, OwnProps, _, _, _, _>(mapStateToProps)(
+  _NeuronInferralModal,
+);
+export const NucleiInferralModal = connect<Props, OwnProps, _, _, _, _>(mapStateToProps)(
+  _NucleiInferralModal,
+);
```

Comment:
Do you have a better Idea to get the dataset and bounding boxes to the modals without code duplication @philippotto?

Currently, it is a mixture of hooks and none-hooks.

Reply:
I'd just switch to hooks all the way. There will still be a tad of duplication probably, but it will be negligible as far I as I see. Just use `useSelector((state: OxalisState) => ...);` for the two store properties each (`dataset` will need to be on the outer components, but `userBoundingBoxes` can be on the inner one, I think). This should make lines 16 to 27 and 279 to 292 obsolete.